api/src/services/listing.service.ts

```diff
@@ -1303,12 +1451,29 @@ export class ListingService implements OnModuleInit {
     // Wrap the deletion and update in one transaction so that units aren't lost if update fails
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const transactions = await this.prisma.$transaction([
-      // delete all connected units before recreating in update
-      this.prisma.units.deleteMany({
-        where: {
-          listingId: storedListing.id,
-        },
-      }),
+      // Conditionally delete either units or unitGroups based on feature flag
+      ...(enableUnitGroups
```

Comment:
nit: Once again, I don't think this case will ever come up in production. But if a listing changes from a jurisdiction that has units and then requires unitGroups the original units wouldn't get deleted. We could potentially just always do all three of these deletes and not put it behind the `enableUnitGroups` flag. But I'd be curious what others think.

Reply:
I updated that, but after changing condition for preventing submitting both units and unitGroups (line: 1583) it should throw an error when such case happen, before transaction.